packages/@aws-cdk/aws-codepipeline-actions/lib/cloudformation/pipeline-actions.ts

```diff
@@ -74,14 +80,28 @@ abstract class CloudFormationAction extends Action {
         maxOutputs: 1,
       },
       inputs,
-      outputs: props.outputFileName
-        ? [props.output || new codepipeline.Artifact(`${props.actionName}_${props.stackName}_Artifact`)]
-        : undefined,
+      outputs: outputFile ? [outputFile.artifact] : undefined,
     });
 
+    this._outputFile = outputFile;
     this.props = props;
   }
 
+  /**
+   * The JSON output file containing the Stack Outputs
+   */
+  public get outputFile(): codepipeline.ArtifactPath | undefined {
+    return this._outputFile;
+  }
+
+  /**
+   * The JSON output file containing the Stack Outputs
+   */
+  public set outputFile(outputFile: codepipeline.ArtifactPath | undefined) {
```

Comment:
Is there a particular reason for this API? The construction properties take in `output?: Artifact` and `outputFileName?: string`, but the setter if for `codepipeline.ArtifactPath`...?